src/views/QuantumTP6/index.js

```diff
@@ -18,25 +18,31 @@ const styles = {
 };
 
 class TP6 extends React.Component {
-  render() {
-    const { classes, location } = this.props;
+  constructor(props) {
+    super(props);
+    const { location } = this.props;
     const params = new URLSearchParams(location.search);
-    const limits = { bits: params.get('bits') };
+    this.state = { bits: params.get('bits') };
+  }
+
+  render() {
+    const { classes } = this.props;
+    const state = this.state;
```

Comment:
If the linting rules were properly working you will get alerted that destructuring is prefered.
```suggestion
    const { bits } = this.state;
```

Reply:
yes, in this case I need the whole `state`, as an object, to send to the `.filter()`